src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/DefaultComesLastCheck.java

```diff
@@ -214,13 +214,11 @@ else if (Objects.nonNull(findNextSibling(defaultGroupAST,
      */
     private static DetailAST findNextSibling(DetailAST ast, int tokenType) {
         DetailAST token = null;
-        DetailAST node = ast.getNextSibling();
-        while (node != null) {
```

Comment:
Did you tried to just remove line 217.
It is optimization to do skip current node, that we can sacrifice in favor of pitest.
Our logic will do loop body for no reason, nano performance degradation.

But pitest gives us more value, so we are ok

Reply:
I have run regression test of changing `node = null`. There was no difference. So I was think maybe we could remove this loop since this function is only used in the DefaultComesLastCheck. The way DefaultComesLastCheck uses this function doesn't require a loop from my perspective. I am not sure which way is better compared to removing line 217.

Reply:
Make this line as `DetailAST node = ast;`

Reply:
Okay. Done.

Reply:
It seems like we can't just change to `DetailAST node = ast;`. The unit test cases where `default` appears as the last one will fail because the `visitToken()` wants to check if any node after the current `default` is switch case or not. Also the name of `findNextSibling()' implies that it will check starting from the next sibling not the current node.
```
switch (i) {
            case 1:
                break;
            case 2:
                break;
            default: // No violation.
                break;
        }
```

Reply:
Keep playing with it, do not give up.
Remove loops, do all you think is reasonable.

Reply:
Done